openmc/settings.py

```diff
@@ -614,17 +613,29 @@ def survival_biasing(self, survival_biasing):
         cv.check_type('survival biasing', survival_biasing, bool)
         self._survival_biasing = survival_biasing
 
-    @weight.setter
-    def weight(self, weight):
-        cv.check_type('weight cutoff', weight, Real)
-        cv.check_greater_than('weight cutoff', weight, 0.0)
-        self._weight = weight
-
-    @weight_avg.setter
-    def weight_avg(self, weight_avg):
-        cv.check_type('average survival weight', weight_avg, Real)
-        cv.check_greater_than('average survival weight', weight_avg, 0.0)
-        self._weight_avg = weight_avg
+    @cutoff.setter
+    def cutoff(self, cutoff):
+        if not isinstance(cutoff, dict):
```

Comment:
I like to keep type-checks a little more general, so in this case use `collections.Mapping` which covers any type that acts like a dictionary.